File: src/main/javabak/tags/ProcessedTags.java
package redcrafter07.processed.tags;

import net.minecraft.resources.ResourceLocation;
import net.minecraft.tags.BlockTags;
import net.minecraft.tags.ItemTags;
import net.minecraft.tags.TagKey;
import net.minecraft.world.item.Item;
import net.minecraft.world.level.block.Block;
import redcrafter07.processed.ProcessedMod;

public class ProcessedTags {
    public static class Items {
        public static final TagKey<Item> FORGE_INGOT_BLITZ = commonTag("ingots/blitz");

        public static TagKey<Item> commonTag(String name) {
            return ItemTags.create(ResourceLocation.fromNamespaceAndPath("c", name));
        }

        public static TagKey<Item> tag(String name) {
            return ItemTags.create(ProcessedMod.rl(name));
        }
    }

    public static class Blocks {
        public static final TagKey<Block> ORE_BLITZ = commonTag("ores/blitz");

        public static TagKey<Block> commonTag(String name) {
            return BlockTags.create(ResourceLocation.fromNamespaceAndPath("c", name));
        }

        public static TagKey<Block> tag(String name) {
            return BlockTags.create(ProcessedMod.rl(name));
        }
    }
}

File: src/main/kotlin/redcrafter07/processed/ProcessedTags.kt
package redcrafter07.processed

import net.minecraft.resources.ResourceLocation
import net.minecraft.tags.BlockTags
import net.minecraft.tags.ItemTags
import net.minecraft.tags.TagKey
import net.minecraft.world.item.Item
import net.minecraft.world.level.block.Block

object ProcessedTags {
    object Items {
        val INGOT_BLITZ = commonTag("ingots/blitz")

        fun commonTag(name: String): TagKey<Item> {
            return ItemTags.create(ResourceLocation.fromNamespaceAndPath("c", name))
        }

        fun tag(name: String): TagKey<Item> {
            return ItemTags.create(rl(name))
        }
    }

    object Blocks {
        val ORE_BLITZ = commonTag("ores/blitz")

        fun commonTag(name: String): TagKey<Block> {
            return BlockTags.create(ResourceLocation.fromNamespaceAndPath("c", name))
        }

        fun tag(name: String): TagKey<Block> {
            return BlockTags.create(rl(name))
        }
    }
}